sweep/tx_input_set.go

```diff
@@ -38,6 +44,15 @@ type InputSet interface {
 
 	// FeeRate returns the fee rate that should be used for the tx.
 	FeeRate() chainfee.SatPerKWeight
+
+	// AddWalletInputs adds wallet inputs to the set until a non-dust
+	// change output can be made. Return an error if there are not enough
+	// wallet inputs.
+	AddWalletInputs(wallet Wallet) error
+
+	// NeedWalletInput returns true if the input set needs more wallet
+	// inputs.
+	NeedWalletInput() bool
```

Comment:
What is the precise relationship between these interface methods?

Reply:
In theory they could be combined into like `MaybeAddWalletInputs`, but I think the goal here is to decouple the two operations so we can check if inputs are needed w/o the main coin selection mutex.

Reply:
yeah we want first do `NeedWalletInput` before holding that lock to decrease the lock scope.